### PR TITLE
Mi/combined cover io UI

### PR DESCRIPTION
**Pull request by mikerooni**
## What
Combines the IO and Cover Configurators on machines into one tab.

## Outcome
- Better UI/UX when configuring cover and auto output settings

## Potential Compatibility Issues
none known

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/api/gui/widget/directional/handlers/CoverableConfigHandler.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            @Override
+            public Component getTitle() {
+                // Uses the widget's own title
+                return Component.empty();
+            }
+
+            @Override
+            public IGuiTexture getIcon() {
+                // TODO make a proper texture for this
+                return new TextTexture("x", Color.BLACK.getRGB());
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/api/gui/widget/directional/handlers/CoverableConfigHandler.java]: DO NOT USE ANYTHING FROM AWT it isn't shipped in some binaries post-java 8

Comment (mikerooni) [reply src/main/java/com/gregtechceu/gtceu/api/gui/widget/directional/handlers/CoverableConfigHandler.java]: Oh thanks lol, didn't realize this was AWT

File: src/main/java/com/gregtechceu/gtceu/api/gui/widget/directional/CombinedDirectionalConfigurator.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    @Override
+    public void handleClientAction(int id, FriendlyByteBuf buf) {
+        if (id != MOUSE_CLICK_CLIENT_ACTION_ID) {
+            super.handleClientAction(id, buf);
+            return;
+        }
+
+        var clickData = ClickData.readFromBuf(buf);
+        var side = buf.readEnum(Direction.class);
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/api/gui/widget/directional/CombinedDirectionalConfigurator.java]: this uses an int. sync a byte of the enum ordinal instead.

File: src/main/java/com/gregtechceu/gtceu/api/gui/widget/directional/CombinedDirectionalConfigurator.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        var result = super.mouseClicked(mouseX, mouseY, button);
+
+        if (isMouseOverElement(mouseX, mouseY) && this.selectedSide == lastSide && this.selectedSide != null) {
+            var hover = sceneWidget.getHoverPosFace();
+
+            if (hover != null && hover.pos.equals(machine.getPos()) && hover.facing == this.selectedSide) {
+                var cd = new ClickData();
+                writeClientAction(MOUSE_CLICK_CLIENT_ACTION_ID, buf -> {
+                    cd.writeToBuf(buf);
+                    buf.writeEnum(this.selectedSide);
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/api/gui/widget/directional/CombinedDirectionalConfigurator.java]: same here

**Review by screret — CHANGES_REQUESTED** [review]

**Review by screret — APPROVED** [review]

**Pull request merged**